### DISCUSSION TITLE
Adding subcatalogs

### DISCUSSION THREAD

**Discussion by glaroc** [stac-utils/stac-fastapi#595] [Q&A]
Hi, I'm wondering if it's possible to add sub catalogs to an existing catalog using stac-fastapi ? I can't seem to find any info on this.

**Comment by gadomski** [comment]
[Answered on twitter (I refuse to call it X)](https://twitter.com/petegadomski/status/1699762653370192057), but short answer, no.

Implementation would be non-trivial, especially if you wanted your sub-catalog to have its own set of STAC API endpoints. Even a basic implementation where the sub-catalog was static (no conformance classes), it would require adding a new out-of-spec endpoint to the server (e.g. `/catalogs/{catalog-id}`), which feels a little funny. There's no reason you couldn't add those endpoints yourself in your own app, but it doesn't feel quite in-scope for this project, to me at least.

I _think_ the "correct" way to go about this would be to create a sub-catalog extension for the [STAC API spec](https://github.com/radiantearth/stac-api-spec), which _then_ could be implemented in **stac-fastapi**. That's much slower, however 😞.

Disclaimer: these are my opinions, others may think differently.

**Reply by glaroc** [reply]
Thank you for the clarification. It is kind of odd that the STAC API spec doesn't have any provision for this, while the STAC spec encourages organizing into subcatalogs. I understand how this would complexify the API however. I'll find another approach in the meantime.

**Reply by vincentsarago** [reply]
>  it would require adding a new out-of-spec endpoint to the server (e.g. /catalogs/{catalog-id}), which feels a little funny

we talk about this in `tipg` (OGC Features services) and technically having `/catalogs/{catalog-id}` is possible and shouldn't break the OGC spec for now, but the OGC Common draft Part 2, totally ignore the `catalog` part and mention only `collection` so maybe the `catalog` will break the spec at one point: https://ogcapi.ogc.org/common/overview.html

Right now, pgstac do not support catalogs either and I'm not sure if it's planned to support it (I guess it would require a lot of work). 


maybe relevant https://github.com/opengeospatial/ogcapi-common/issues/11 https://github.com/opengeospatial/ogcapi-common/issues/298

**Reply by gadomski** [reply]
> It is kind of odd that the STAC API spec doesn't have any provision for this, while the STAC spec encourages organizing into subcatalogs.

I agree. There's some fundamental differences between static catalogs and "API catalogs", and we don't tell that story particularly well at the moment (IMO). This is a topic that we'll be working on, at least from the documentation/story-telling side, during the upcoming [STAC sprint](https://cloudnativegeo.org/blog/2023/07/stac-sprint-8).
